### ISSUE TITLE
Missing network-manager.service systemd unit

### ISSUE THREAD

**Issue by philandstuff** [NixOS/nixpkgs#15552]
## Issue description

I upgraded NixOS recently, and NetworkManager no longer starts. It seems the network-manager.service unit has gone missing somehow, so nothing triggers to start NetworkManager.

16.09pre82105.7465bcd was a version which worked correctly.
16.09pre83796.d541e0d fails with the same configuration.nix file.

I've traced it down to the `/run/current-version/etc/systemd/system/network-manager.service` symlink disappearing between the versions, even though the NetworkManager package is still installed.

## Technical details
(run `nix-env --version`)
* Nix version: 1.11.2
* Nixpkgs version: from NixOS

**Comment by bjornfor** [comment]
Still (git) bisecting... (By the speed of the bisecting, or lack thereof, tomorrow I might have to try *understanding* how this thing works, instead of relying on dumb bisecting.)

**Comment by bjornfor** [comment]
A "fun" thing when bisecting is that the system closure must be _very_ small to have a chance of building successfully. There is a lot of breakage on master :-(

**Comment by dezgeg** [comment]
Taking a wild guess: it's due to the multiple-outputs split of NetworkManager. Does this help?

````
diff --git a/nixos/modules/services/networking/networkmanager.nix b/nixos/modules/services/networking/networkmanager.nix
index e9eea6a..e17d77b 100644
--- a/nixos/modules/services/networking/networkmanager.nix
+++ b/nixos/modules/services/networking/networkmanager.nix
@@ -240,7 +240,7 @@ in {
       uid = config.ids.uids.avahi-autoipd;
     }];
 
-    systemd.packages = cfg.packages;
+    systemd.packages = cfg.packages.out;
 
     # Create an initialisation service that both starts
     # NetworkManager when network.target is reached,
````

**Comment by philandstuff** [comment]
@dezgeg nice guess!

The following line in my `/etc/nixos/configuration.nix` works around the problem for the moment:

```
  systemd.packages = [ pkgs.networkmanager.out ];
```

I'm now successfully running 16.09pre83796.d541e0d.

**Comment by bjornfor** [comment]
@dezgeg: Thanks for the hint! I tried your patch and found that it doesn't (directly) work because cfg.packages is a list. The cfg.packages list is used multiple places in the nixos module. So the question is whether to modify the list directly, by appending .out to each element, or something else. I don't know whether it's safe to assume that all packages in cfg.packages have their "useful" output in .out...

@philandstuff: Good to know. The question is, what about all of networkmanager's plugins? Should we add .out to all cfg.packages?

**Comment by bjornfor** [comment]
The current multiple-outputs split for NetworkManager saves 3 MiB (of 16 MiB). Because NetworkManager is currently broken, and the fact that there are several ways to fix it going forward (and I'm not sure which way is the best), I plan to revert the multiple-outputs change for NetworkManager. If someone wants those 3 MiB back, just make a (complete) patch.

CC @ttuegel, @abbradar

**Comment by bjornfor** [comment]
Fixed/reverted in c7db50e24f2a3522abd98da7e605a06002e5cdba (and two following parents).

**Comment by philandstuff** [comment]
Thanks, @bjornfor!
